app/components/Tooltip/index.tsx

```diff
@@ -9,127 +9,72 @@ type Props = {
   className?: string;
   onClick?: () => void;
   style?: CSSProperties;
-  list?: boolean;
-  renderDirection?: string;
-  pointerPosition?: string;
 };
-type State = {
-  hovered: boolean;
-  childrenContainerWidth: number;
-};
-/**
- * A tooltip that appears when you hover over the component placed within.
- * The tooltip will by default be centered, however it supports a 'renderDirection'
- * prop that will make it render either to the left or the right from the postion of the pointer. The pointer
- * (the small arrow that points towards the component within the tooltip) will
- * also default to center, and it can be adjusted with the 'pointerPosition' prop.
- * Both props can be set as either 'left' or 'right'.
- */
-
-export default class Tooltip extends Component<Props, State> {
-  static defaultProps = {
-    list: false,
-  };
-  tooltip: HTMLDivElement | null | undefined;
-
-  measure() {
-    if (!this.tooltip) {
-      return;
-    }
-
-    const width = this.tooltip.offsetWidth;
-    this.setState({
-      childrenContainerWidth: width,
-    });
-  }
-
-  state = {
-    hovered: false,
-    childrenContainerWidth: 0,
-  };
-  onMouseEnter = () => {
-    this.setState({
-      hovered: true,
-    });
-  };
-  onMouseLeave = () => {
-    this.setState({
-      hovered: false,
-    });
-  };
-
-  componentDidMount() {
-    this.measure();
-  }
 
-  render() {
-    const {
-      content,
-      children,
-      className,
-      list,
-      style,
-      onClick,
-      renderDirection,
-      pointerPosition,
-    } = this.props;
-    let renderDirectionClass = styles.renderFromCenter;
-    let startPointChildren = 2;
+const Tooltip = ({ children, content, className, style, onClick }: Props) => {
+  const [hovered, setHovered] = useState(false);
+  const tooltipRef = useRef<HTMLDivElement | null>(null);
+  const triggerRef = useRef<HTMLDivElement | null>(null);
+  const arrowRef = useRef<HTMLDivElement | null>(null);
```

Comment:
```suggestion
  const tooltipRef = useRef<HTMLDivElement>(null);
  const triggerRef = useRef<HTMLDivElement>(null);
  const arrowRef = useRef<HTMLDivElement>(null);
```
Don't need the null in type arg here

Reply:
Good point